kroxylicious-filters/kroxylicious-record-validation/src/main/java/io/kroxylicious/proxy/filter/validation/validators/bytebuf/JwsSignatureBytebufValidator.java

```diff
@@ -0,0 +1,163 @@
+ */
+
+package io.kroxylicious.proxy.filter.validation.validators.bytebuf;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.record.Record;
+import org.jose4j.jwa.AlgorithmConstraints;
+import org.jose4j.jwk.JsonWebKey;
+import org.jose4j.jwk.JsonWebKeySet;
+import org.jose4j.jwk.VerificationJwkSelector;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.lang.JoseException;
+import org.jose4j.lang.UnresolvableKeyException;
+
+import io.kroxylicious.proxy.config.tls.AllowDeny;
+import io.kroxylicious.proxy.filter.validation.validators.Result;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Checks if the {@link Record} headers contains a valid {@link JsonWebSignature} Signature.
+ * <p>
+ * The JWS Signature is validated using the provided {@link JsonWebKeySet} and {@link AlgorithmConstraints}.
+ * </p>
+ *
+ * @see <a href="https://datatracker.ietf.org/doc/html/rfc7515">RFC 7515 (JWS)</a>
+ * @see <a href="https://datatracker.ietf.org/doc/html/rfc7797">RFC 7797 (JWS Unencoded Payload Option)</a>
+ * @see <a href="https://datatracker.ietf.org/doc/html/rfc7517">RFC 7517 (JWK)</a>
+ * @see <a href="https://datatracker.ietf.org/doc/html/rfc7518">RFC 7518 (JWA)</a>
+ */
+public class JwsSignatureBytebufValidator implements BytebufValidator {
+    private static final String DEFAULT_ERROR_MESSAGE = "JWS Signature could not be successfully verified";
+    private final String jwsRecordHeaderKey;
+
+    private static final VerificationJwkSelector jwkSelector = new VerificationJwkSelector();
+
+    private final JsonWebSignature jws;
+    private final JsonWebKeySet trustedJsonWebKeySet;
+    private final boolean isContentDetached;
+    private final boolean failOnMissingJwsRecordHeader;
+
+    /**
+     * Constructor for {@link JwsSignatureBytebufValidator}.
+     *
+     * @see <a href="https://bitbucket.org/b_c/jose4j/wiki/JWS%20Examples">jose4j JWS examples</a>
+     */
+    public JwsSignatureBytebufValidator(JsonWebKeySet trustedJsonWebKeySet, AllowDeny<String> allowedAndDeniedAlgorithms, String jwsRecordHeaderKey,
+                                        boolean isContentDetached, boolean failOnMissingJwsRecordHeader) {
+        this.jws = new JsonWebSignature();
+        this.trustedJsonWebKeySet = trustedJsonWebKeySet;
+        this.jwsRecordHeaderKey = jwsRecordHeaderKey;
+        this.isContentDetached = isContentDetached;
+        this.failOnMissingJwsRecordHeader = failOnMissingJwsRecordHeader;
+
+        AlgorithmConstraints algorithmConstraints = extractAlgorithmConstraints(allowedAndDeniedAlgorithms);
```

Comment:
Personally I'd inject the `algorithmConstraints` rather parsing in the ctor.

Reply:
Inject in the sense used in Inversion Of Control (IOC) / dependency injection. Pass the AlgorithmConstraints as an argument to this constructor rather than `allowedAndDeniedAlgorithms`. 

Though the more a look at the validators thats going to stand out like a sore thumb. Lets just leave things as they are for now.